src/transformation/builtins/function.ts

```diff
@@ -40,7 +40,11 @@ export function transformFunctionProperty(
 ): lua.Expression | undefined {
     switch (node.name.text) {
         case "length":
-            if (context.luaTarget === LuaTarget.Lua51 || context.luaTarget === LuaTarget.Universal) {
+            if (
+                context.luaTarget === LuaTarget.Lua50 ||
+                context.luaTarget === LuaTarget.Lua51 ||
+                context.luaTarget === LuaTarget.Universal
+            ) {
                 context.diagnostics.push(unsupportedForTarget(node, "function.length", LuaTarget.Lua51));
```

Comment:
Noticed a small bug in our code, could you take it along: I think this diagnostic should actually get `context.luaTarget` instead of hardcoding `LuaTarget.Lua51`

Reply:
Fixed with 635d271841dd637f8ef44f4088c36fdd9df2ed7f!